### PR TITLE
Updated Deck class

### PR DESCRIPTION
**Pull request by GEvanscsumb**
Deck class has been improved and comments added.

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+</projectDescription>
```
Comment (tramstheman) [inline .project]: please add this to the gitignore and remove it from the commit

File: .classpath [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="output" path="bin"/>
+</classpath>
```
Comment (tramstheman) [inline .classpath]: please add this to the gitignore and remove it from the commit

File: src/com/northvine/engineering/cards/Deck.java [hunk 3]
```diff
@@ -1,13 +1,173 @@
...
+    * 
+    * allocateMasterPack creates the standard Deck that the public variable cards
+    * will reference when init is called.
+    */
+   public Deck()
+   {
+      allocateMasterPack();
+      numPacks = 1;
+      init(numPacks);
+   }
```
Comment (tramstheman) [inline src/com/northvine/engineering/cards/Deck.java]: init(numPacks = 1);

File: src/com/northvine/engineering/cards/Deck.java [hunk 4]
```diff
@@ -1,13 +1,173 @@
...
+         for (char legalValue : value)
+         {
+            masterPack[i] = new Card(legalValue, Card.Suit.HEARTS);
+            i++;
+         }
+         for (char legalValue : value)
+         {
+            masterPack[i] = new Card(legalValue, Card.Suit.SPADES);
+            i++;
+         }
```
Comment (tramstheman) [inline src/com/northvine/engineering/cards/Deck.java]: This can be done in a single for loop.

File: src/com/northvine/engineering/Main.java [hunk 5]
```diff
@@ -106,4 +108,44 @@ private static void cardTestDriver()
...
+    * Declare a deck containing two packs of cards. Do not shuffle.  
+    * Deal all the cards in a loop until the deck is empty (dealt directly to the display/screen, not to any Hand objects just yet).  
+    * Display each card as it comes off the deck.  Next, reset the deck by initializing it again (to the same two packs).  
+    * Shuffle the deck this time, and re-deal to the screen in a loop again. Notice that the cards are now coming off in a random order.
+    * Repeat this double deal, unshuffled, then shuffled, but this time using a single pack deck.
+    */
+   private static void deckTestDriver()
+   {
+      System.out.println("\nTesting Deck class with paramter of 2.\n");
+      Deck myDeck = new Deck();
```
Comment (akatrent) [inline src/com/northvine/engineering/Main.java]: Pass in 2 here.  Deck myDeck = new Deck(2);

**Pull request merged**